hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Metric.java

```diff
@@ -16,175 +16,13 @@
-            this(metricTypePath, id, Collections.emptyMap());
-        }
-
-        public Blueprint(String metricTypePath, String id, Map<String, Object> properties) {
-            super(id, properties);
-            this.metricTypePath = metricTypePath;
-            this.collectionInterval = null;
-        }
-
-        public Blueprint(String metricTypePath, String id, Map<String, Object> properties,
-                         Map<String, Set<CanonicalPath>> outgoing,
-                         Map<String, Set<CanonicalPath>> incoming) {
-            super(id, properties, outgoing, incoming);
-            this.metricTypePath = metricTypePath;
-            this.collectionInterval = null;
-        }
-
-        public Blueprint(String metricTypePath, String id, String name, Map<String, Object> properties,
-                         Map<String, Set<CanonicalPath>> outgoing,
-                         Map<String, Set<CanonicalPath>> incoming) {
-            super(id, name, properties, outgoing, incoming);
-            this.metricTypePath = metricTypePath;
-            this.collectionInterval = null;
-        }
-
-        public Blueprint(String metricTypePath, String id, String name, Long collectionInterval,
-                         Map<String, Object> properties,
-                         Map<String, Set<CanonicalPath>> outgoing,
-                         Map<String, Set<CanonicalPath>> incoming) {
-            super(id, name, properties, outgoing, incoming);
-            this.metricTypePath = metricTypePath;
-            this.collectionInterval = collectionInterval;
-        }
-
-        public String getMetricTypePath() {
-            return metricTypePath;
-        }
-
-        public Long getCollectionInterval() {
-            return collectionInterval;
-        }
-
-        @Override
-        public <R, P> R accept(ElementBlueprintVisitor<R, P> visitor, P parameter) {
-            return visitor.visitMetric(this, parameter);
-        }
-
-        public static final class Builder extends Entity.Blueprint.Builder<Blueprint, Builder> {
-            private String metricTypeId;
-            private Long collectionInterval;
-
-            public Builder withMetricTypePath(String metricTypePath) {
-                this.metricTypeId = metricTypePath;
-                return this;
-            }
-
-            public Builder withInterval(Long interval) {
-                this.collectionInterval = interval;
-                return this;
-            }
-
-            @Override
-            public Blueprint build() {
-                return new Blueprint(metricTypeId, id, name, collectionInterval, properties, outgoing, incoming);
-            }
-        }
-    }
+public class Metric {
+    private String id;
+    private String name;
+    private String feed;    // Index; But not sure if feeds are still in play if the inventory is built from directly prometheus scans
+    private MetricType type;  // Index [Search all metrics of type xx]
```

Comment:
@lucasponce here the ideal index would be "metric.type.id" (for query such as "search all metrics whose type id is xx")
Is it easy to do?

Reply:
Yeah, I think that perhaps MetricType is stored separately from Metric, then, here we could have a typeId that serves as the link with the MetricType and also it is a field for indexing.